packages/nimble-components/src/rich-text/specs/mention-hld.md

```diff
@@ -0,0 +1,706 @@
+Within its default slot, it gets `nimble-list-option` elements which is used to populate the list with the provided options.
+
+This component is responsible for updating the options property by filtering the list of options provided by the parent component, which in this context is the
+`nimble-rich-text-editor`. When the options are updated, the component internally handles setting the selected index values within the list-box.
+To facilitate keyboard interactions, a public method called `keyDownHandler()` is provided in `nimble-rich-text-mention-list-box`. This method is invoked in the
+`onKeyDown()` function in the Tiptap configuration, as described in the [configuration on Tiptap](#1-configurations-on-tiptap) section. The `keyDownHandler()`
+method takes the keyboard event as input and performs necessary operations, such as responding to up and down arrow key presses to decrement or increment the
+selected index value for each option, and handling `Enter` key presses for option selection, and so on.
+
+A click handler is attached to the `nimble-list-box`, and it emits the selected option data to the parent component. The parent component then invokes the `mention.command`
+from Tiptap to render the mention node based on the selected option data.
+
+Both the click and key down handlers will emit a `change` event which is to be listened by the parent element to render the mention node based on selection in the editor.
+
+### Angular integration
+
+An Angular directive will be created for `nimble-rich-text-mention-users` and `nimble-mapping-mention-user` components. Input and accessor APIs will be created for the attributes
+and properties and output event emitters will be created for the events, similar to how it's done in other directives. The components will not have form
+association, so a `ControlValueAccessor` will not be created.
+
+### States
+
+When you add **"@"** into the editor, a dropdown will open with the first value as the selected option, indicated by a green overlay to the option
+if `nimble-list-option` is passed into the default slot element. Upon selecting an option from the list, it will be rendered as prominent green text
+with bold formatted by default in all themes. Below is a basic representation of `@mention` texts and dropdown lists in the default theme.
+
+![@mention State](./spec-images/mention-state.png)
+
+An **"@"** icon will be added to the formatting toolbar to insert the **"@"** character, and clicking the button will open a dropdown list of options.
+This button will not be highlighted like other buttons when the cursor is placed on the @mentions within the editor.
+
+### Accessibility
+
+_Focus_
+
+-   Focus state of the list options for `@mention` will be the same as the `nimble-list-option`.
+-   Focusing out from the editor will close the `@mention` popup if it is already open.
```

Comment:
What about other actions like scrolling within the editor or scrolling the page so that the mention is no longer on screen? Will the `@mention` popup remain visible? Will it move with the thing it's anchored to?

Reply:
When scrolling the entire page the popup will move along with the `@` but scrolling the editor will not move the popup and will stay in the same place where it was originally opened. We could see the same behavior in AzDo as well as in the GitHub. I have updated the same in the HLD as well.